### PR TITLE
Teet 117

### PR DESCRIPTION
**Pull request by kuroneko1011**

### CODE REVIEW

File: src/components/customized-components/ClinicCard/ClinicCard.component.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { Button } from "antd";
+import { EnvironmentFilled, StarFilled } from "@ant-design/icons";
+import "./ClinicCard.style.scss";
+
+const ClinicCardComponent = (props) => {
```
Comment (ZERO-94) [inline src/components/customized-components/ClinicCard/ClinicCard.component.jsx:6]: change this into {imgSrc, name, serviceArray,...} or pass an object into this, props is confusing

File: src/components/customized-components/ClinicCard/ClinicCard.component.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        <div className="card-home-page-rating">
+          <div>{props.avgRatingScore}</div>
+          <StarFilled style={{ color: "#FFCB45" }} />
+        </div>
+        <div className="card-clinic-clinic-page-button">
+          <Button
+            type="primary"
+            shape="round"
+            size="large"
+            onClick={() => props.onClick(props.id)}
```
Comment (ZERO-94) [inline src/components/customized-components/ClinicCard/ClinicCard.component.jsx:34]: why this onClick is already in that data but still need to pass the id into the function ????

File: src/components/customized-components/ClinicDetailComponent/ClinicDetail.component.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+// import {}
+
+const ClinicDetailComponent = (props) => {
```
Comment (ZERO-94) [inline src/components/customized-components/ClinicDetailComponent/ClinicDetail.component.jsx:4]: change this into {imgSrc, name, serviceArray,...} or pass an object into this, props is confusing

File: src/components/customized-components/HomePageContentCard/HomePageContentCard.component.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import "./HomePageContentCard.style.css";
+
+const HomePageContentCard = ({ context }) => {
```
Comment (ZERO-94) [inline src/components/customized-components/HomePageContentCard/HomePageContentCard.component.jsx:4]: HomePageContentCard"Component"

File: src/components/customized-components/HomePageContentCardPreview/HomePageContentCardPreview.component.jsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import cardHomePage1 from "../../../assets/cardImg1.png";
+import cardHomePage2 from "../../../assets/cardImg2.png";
+import cardHomePage3 from "../../../assets/cardImg3.png";
+
+import HomePageContentCard from "../HomePageContentCard/HomePageContentCard.component";
+
+import "./HomePageContentCardPreview.style.css";
+
+const HomePageContentCardPreview = () => {
```
Comment (ZERO-94) [inline src/components/customized-components/HomePageContentCardPreview/HomePageContentCardPreview.component.jsx:10]: HomePageContentCardPreview"Component"

File: src/components/customized-components/IntroductionPart/IntroductionPart.component.jsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { Col, Row, Button } from "antd";
+import { SearchOutlined } from "@ant-design/icons";
+
+import "./IntroductionPart.style.css";
+
+const IntroductionPart = ({ homePageImg }) => {
```
Comment (ZERO-94) [inline src/components/customized-components/IntroductionPart/IntroductionPart.component.jsx:7]: IntroductionPart"Component"

File: src/components/customized-components/LocationSelectBlock/LocationSelectBlock.component.jsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { Form, Select } from "antd";
+import { ClinicRegisterValidation } from "../../../validate/RegisterValidation";
+
+const LocationSelectBlock = (props) => {
```
Comment (ZERO-94) [inline src/components/customized-components/LocationSelectBlock/LocationSelectBlock.component.jsx:5]: Location"InputComponent"

Comment (ZERO-94) [inline src/components/customized-components/LocationSelectBlock/LocationSelectBlock.component.jsx:5]: {provinces, districts, wards,...} not props only

File: src/components/forms/ClinicRegisterForm/ClinicRegisterForm.component.jsx [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+import { Form, Input, Button, Select, Row, Col } from "antd";
+import {
+  AttendantRegisterValidation,
+  ClinicRegisterValidation,
+} from "../../../validate/RegisterValidation";
+
+import LocationContainer from "../../../containers/LocationContainer/LocationContainer.container";
+import { GenderType } from "../../../constants/GenderConstants";
+
+const ClinicRegisterForm = (props) => {
```
Comment (ZERO-94) [inline src/components/forms/ClinicRegisterForm/ClinicRegisterForm.component.jsx:12]: ClinicRegisterForm"Component"

File: src/components/forms/UserRegisterForm/UserRegisterForm.component.jsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import { Form, Input, Button, Select } from "antd";
+
+import { AttendantRegisterValidation } from "../../../validate/RegisterValidation";
+import { GenderType } from "../../../constants/GenderConstants";
+
+import "./UserRegisterForm.style.css";
+
+const UserRegisterForm = (props) => {
```
Comment (ZERO-94) [inline src/components/forms/UserRegisterForm/UserRegisterForm.component.jsx:9]: UserRegisterForm"Component"

File: src/services/teeth-apis/RegisterController.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
+import { CallAPI } from "./axiosBase";
+
+export const patientRegisterAPI = (values) =>
+  CallAPI("/patients", "POST", values);
+
+export const managerRegisterAPI = (values) =>
```
Comment (ZERO-94) [inline src/services/teeth-apis/RegisterController.js:6]: same as above

**Review by ZERO-94 — CHANGES_REQUESTED** [review]

File: src/containers/ClinicContainer/ClinicContainer.container.jsx [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+      } catch (e) {
+        notification["error"]({
+          message: e,
+          duration: 2,
+        });
+      }
+    };
+    getService();
+    getClinic();
+    onFinish();
```
[the remark this answers is not in the document]
Comment (kuroneko1011) [reply src/containers/ClinicContainer/ClinicContainer.container.jsx:58]: For filter with no params for when loading the page :)  to display all clinics

File: src/components/customized-components/ClinicCard/ClinicCard.component.jsx [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+        <div className="card-home-page-rating">
+          <div>{clinic.avgRatingScore}</div>
+          <StarFilled style={{ color: "#FFCB45" }} />
+        </div>
+        <div className="card-clinic-clinic-page-button">
+          <Button
+            type="primary"
+            shape="round"
+            size="large"
+            onClick={() => onClick(clinic.id)}
```
Comment (ZERO-94) [inline src/components/customized-components/ClinicCard/ClinicCard.component.jsx:39]: fix this

File: src/components/customized-components/ClinicDetailComponent/ClinicDetail.component.jsx [hunk 13]
```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import { StarOutlined, EnvironmentOutlined } from "@ant-design/icons";
+import { Button, Card } from "antd";
+import "./ClinicDetail.style.scss";
+import avtImg from "../../../assets/clinicImg.png";
+
+const ClinicDetailComponent = ({ currentClinic }) => {
```
Comment (ZERO-94) [inline src/components/customized-components/ClinicDetailComponent/ClinicDetail.component.jsx:7]: clinicData

Comment (ZERO-94) [inline src/components/customized-components/ClinicDetailComponent/ClinicDetail.component.jsx:7]: clinicData

File: src/components/customized-components/HomePageContentCardPreview/HomePageContentCardPreview.component.jsx [hunk 14]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import cardHomePage1 from "../../../assets/cardImg1.png";
+import cardHomePage2 from "../../../assets/cardImg2.png";
+import cardHomePage3 from "../../../assets/cardImg3.png";
+
+import HomePageContentCardComponent from "../HomePageContentCard/HomePageContentCard.component";
+
+import "./HomePageContentCardPreview.style.css";
+
+const HomePageContentCardPreviewComponent = () => {
```
Comment (ZERO-94) [inline src/components/customized-components/HomePageContentCardPreview/HomePageContentCardPreview.component.jsx:10]: HomPagePreviewContentCard

Comment (ZERO-94) [inline src/components/customized-components/HomePageContentCardPreview/HomePageContentCardPreview.component.jsx:10]: HomPagePreviewContentCard

File: src/components/customized-components/LocationInput/LocationInput.component.jsx [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+  selectedValue,
+  handleValueChange,
+}) => {
+  const { Option } = Select;
+  return (
+    <>
+      <Form.Item
+        name={name}
+        label={label}
+        // rules={ClinicRegisterValidation.province}
```
Comment (ZERO-94) [inline src/components/customized-components/LocationInput/LocationInput.component.jsx:19]: add comment TODO filter on container

Comment (ZERO-94) [inline src/components/customized-components/LocationInput/LocationInput.component.jsx:19]: add comment TODO filter on container

File: src/components/forms/UserRegisterForm/UserRegisterForm.component.jsx [hunk 16]
```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { Form, Input, Button, Select } from "antd";
+
+import { AttendantRegisterValidation } from "../../../validate/RegisterValidation";
+import { GenderType } from "../../../constants/GenderConstants";
+
+import "./UserRegisterForm.style.css";
+
+const UserRegisterFormComponent = ({ onFinish }) => {
```
Comment (ZERO-94) [inline src/components/forms/UserRegisterForm/UserRegisterForm.component.jsx:9]: Patient or user?

Comment (ZERO-94) [inline src/components/forms/UserRegisterForm/UserRegisterForm.component.jsx:9]: Patient or user?

File: src/containers/ClinicCardContainer/ClinicCardContainer.container.jsx [hunk 17]
```diff
@@ -0,0 +1,99 @@
+import React, { Fragment, useState } from "react";
```
Comment (ZERO-94) [inline src/containers/ClinicCardContainer/ClinicCardContainer.container.jsx:1]: fix file naming!

Comment (ZERO-94) [inline src/containers/ClinicCardContainer/ClinicCardContainer.container.jsx:1]: fix file naming!

File: src/containers/ClinicCardContainer/ClinicCardContainer.container.jsx [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+import { getClinicsAPI } from "../../services/teeth-apis/ClinicController";
+
+import ClinicCardComponent from "../../components/customized-components/ClinicCard/ClinicCard.component";
+import "./ClinicCardContainer.style.scss";
+
+const ClinicCardContainer = ({ clinicData, layoutDirection }) => {
+  const history = useHistory();
+
+  const onClick = (idClinic) => {
+    history.push(RoutePath.CLINIC_PAGE + "/" + idClinic);
```
Comment (ZERO-94) [inline src/containers/ClinicCardContainer/ClinicCardContainer.container.jsx:18]: fix this

Comment (ZERO-94) [inline src/containers/ClinicCardContainer/ClinicCardContainer.container.jsx:18]: fix this

File: src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx [hunk 19]
```diff
@@ -0,0 +1,122 @@
...
+import { notification, Form, Button, Select } from "antd";
+import ClinicCardContainer from "../ClinicCardContainer/ClinicCardContainer.container";
+import { filterClinicAPI } from "../../services/teeth-apis/ClinicController";
+import { getAllServices } from "../../services/teeth-apis/ServiceController";
+import { getClinicsAPI } from "../../services/teeth-apis/ClinicController";
+import LocationInputContainer from "../LocationInputContainer/LocationInputContainer.container";
+
+import "./ClinicPreviewContainer.style.scss";
+
+const ClinicPreviewContainer = () => {
```
Comment (ZERO-94) [inline src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx:11]: Preview what ?

Comment (ZERO-94) [inline src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx:11]: Preview what ?

File: src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx [hunk 20]
```diff
@@ -0,0 +1,122 @@
...
+    } catch (e) {
+      notification["error"]({
+        message: `Something went wrong! Try again latter!`,
+        description: `There is problem while filter, try again later`,
+        duration: 2,
+      });
+    }
+  };
+
+  const getService = async () => {
```
Comment (ZERO-94) [inline src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx:33]: fetch not get

Comment (ZERO-94) [inline src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx:33]: fetch not get

File: src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx [hunk 21]
```diff
@@ -0,0 +1,122 @@
...
+      setServices(data.content);
+    } catch (e) {
+      notification["error"]({
+        message: e,
+        duration: 2,
+      });
+    }
+  };
+
+  const getClinic = async () => {
```
Comment (ZERO-94) [inline src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx:45]: fetch or get ?

Comment (ZERO-94) [inline src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx:45]: fetch or get ?

File: src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx [hunk 22]
```diff
@@ -0,0 +1,122 @@
...
+
+  const [clinics, setClinics] = useState([]);
+
+  const [services, setServices] = useState([]);
+  const [filteredClinic, setFilteredClinic] = useState(clinics);
+  const [selectedService, setSelectedService] = useState();
+
+  const onFinish = async (values) => {
+    try {
+      const { data } = await filterClinicAPI(values);
```
Comment (ZERO-94) [inline src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx:22]: no need to make 2 functions for filter

Comment (ZERO-94) [inline src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx:22]: no need to make 2 functions for filter

File: src/containers/LocationInputContainer/LocationInputContainer.container.jsx [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+    setSelectedProvince(e);
+  };
+  const handleDistrictChange = (e) => {
+    setSelectedDistrict(e);
+  };
+  const handleWardChange = (e) => {
+    setSelectedWard(e);
+  };
+
+  return (
```
Comment (ZERO-94) [inline src/containers/LocationInputContainer/LocationInputContainer.container.jsx:47]: add TODO comment to optimize

Comment (ZERO-94) [inline src/containers/LocationInputContainer/LocationInputContainer.container.jsx:47]: add TODO comment to optimize

File: src/components/forms/ClinicRegisterFormComponent/ClinicRegisterForm.component.jsx [hunk 24]
```diff
@@ -0,0 +1,132 @@
...
+} from "../../../validate/RegisterValidation";
+
+import LocationInputContainer from "../../../containers/LocationInputContainer/LocationInputContainer.container";
+import { GenderType } from "../../../constants/GenderConstants";
+
+const ClinicRegisterFormComponent = (props) => {
+  const { Option } = Select;
+
+  return (
+    <Form
```
Comment (ZERO-94) [inline src/components/forms/ClinicRegisterFormComponent/ClinicRegisterForm.component.jsx:16]: add TODO to useForm next time

Comment (ZERO-94) [inline src/components/forms/ClinicRegisterFormComponent/ClinicRegisterForm.component.jsx:16]: add TODO to useForm next time

**Review by ZERO-94 — CHANGES_REQUESTED** [review]

File: src/components/customized-components/ClinicCard/ClinicCard.component.jsx (hunk 12 above)
Comment (kuroneko1011) [reply src/components/customized-components/ClinicCard/ClinicCard.component.jsx:39]: fixed in TEET 86

File: src/components/customized-components/ClinicDetailComponent/ClinicDetail.component.jsx (hunk 13 above)
Comment (kuroneko1011) [reply src/components/customized-components/ClinicDetailComponent/ClinicDetail.component.jsx:7]: fix in TEET 86

Comment (kuroneko1011) [reply src/components/customized-components/ClinicDetailComponent/ClinicDetail.component.jsx:7]: also fixed in TEET 86

File: src/components/customized-components/HomePageContentCardPreview/HomePageContentCardPreview.component.jsx (hunk 14 above)
Comment (kuroneko1011) [reply src/components/customized-components/HomePageContentCardPreview/HomePageContentCardPreview.component.jsx:10]: it is a list of HomePageContentCard. Maybe I change it into HomePageContentCardList ?

Comment (kuroneko1011) [reply src/components/customized-components/HomePageContentCardPreview/HomePageContentCardPreview.component.jsx:10]: mention above

File: src/components/customized-components/LocationInput/LocationInput.component.jsx (hunk 15 above)
Comment (kuroneko1011) [reply src/components/customized-components/LocationInput/LocationInput.component.jsx:19]: done

File: src/components/forms/UserRegisterForm/UserRegisterForm.component.jsx (hunk 16 above)
Comment (kuroneko1011) [reply src/components/forms/UserRegisterForm/UserRegisterForm.component.jsx:9]: fixed in TEET 86

File: src/containers/ClinicCardContainer/ClinicCardContainer.container.jsx (hunk 17 above)
Comment (kuroneko1011) [reply src/containers/ClinicCardContainer/ClinicCardContainer.container.jsx:1]: done

File: src/containers/ClinicCardContainer/ClinicCardContainer.container.jsx (hunk 18 above)
Comment (kuroneko1011) [reply src/containers/ClinicCardContainer/ClinicCardContainer.container.jsx]: fixed in TEET 86

File: src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx (hunk 20 above)
Comment (kuroneko1011) [reply src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx]: fixed in TEET 86

File: src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx (hunk 22 above)
Comment (kuroneko1011) [reply src/containers/ClinicPreviewContainer/ClinicPreviewContainer.container.jsx]: fixed in TEET 86

File: src/containers/LocationInputContainer/LocationInputContainer.container.jsx (hunk 23 above)
Comment (kuroneko1011) [reply src/containers/LocationInputContainer/LocationInputContainer.container.jsx:47]: done

**Pull request closed without merging**